packages/@lwc/engine-core/src/framework/base-lightning-element.ts

```diff
@@ -509,7 +517,7 @@ LightningElement.prototype = {
         return getClassList(elm);
     },
 
-    get template(): ShadowRoot {
+    get template(): ShadowRoot | null {
         const vm = getAssociatedVM(this);
         return vm.cmpRoot;
```

Comment:
I would be interesting to add a warning message in DEV mode when accessing the `template` property when a component is running in the Light DOM.